### DISCUSSION TITLE
How to use rootless docker-compose, podman play kube, and the podman-kube@.service?

### DISCUSSION THREAD

**Discussion by andrew-kennedy** [containers/podman#15951] [Q&A]
Here's my scenario:

I have several services defined and configured in separate docker-compose.yml files. I want to turn those files into rootless running services, and then into kube yaml, and then run that kube yaml on system boot through the `podman-kube@.service` template. I don't want to have to specify any manual port-forwards/mappings, I want to be able to resolve services by their `container-name:port` within the same network namespace. So one service running a reverse proxy should be able to reach another service's web ui at `servicename:8443` directly. A few questions:

1. I currently notice that docker-compose services launched separately from different files with different invocations of `docker-compose up -d` do not seem to be able to resolve each other by name if I join them to an external network I create in advance. Is this because DNS doesn't work in non-default networks with slirp4netns? Is there a way around this? (and is there a way around this that keeps original inbound IP-addresses like the `slirp4netns` port handler?)
2. I noticed that it seems special network configuration or pod grouping is eliminated when launching through a play kube with a generated kubernetes yaml from a `docker-compose.yml` service. Is there a way to generate many kubernetes service yaml descriptions and when each is separately launched via `podman play kube` via the `podman-kube@.service` template, have them all operating within the same network namespace?

**Comment by vrothberg** [comment]
Thanks for reaching out, @andrew-kennedy!

@rhatdan @umohnani8 @baude @mheon PTAL

**Comment by mheon** [comment]
Podman version is very important here. Can you provide a full `podman info`?
